xdc-plugin/xdc.cc

```diff
@@ -126,7 +130,28 @@ struct SetProperty : public Pass {
         if (design->top_module() == nullptr) {
             log_cmd_error("No top module detected\n");
         }
+        if (args.at(1) == "-dict") {
+            std::string dict_args = args.at(2).c_str();
+            trim(dict_args);
+            std::stringstream args_stream(dict_args);
+            std::vector<std::string> tokens;
+            std::string intermediate;
+            while (getline(args_stream, intermediate, ' ')) {
+                tokens.push_back(intermediate);
+            }
+            for (long unsigned int i = 0; i < tokens.size(); i += 2) {
```

Comment:
We should probably have a sanity check if the number of tokens is even and error out if this is not the case.

Reply:
Added a validation. I had written the code but forgot to commit it! :)